File: ruby/src/system.rb
############################
# Copyright 2011 Thomas SCHWEBEL
# 
# This file is part of The Mercury Ruby SDK.
# 
# The Mercury Ruby SDK is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# 
# The Mercury Ruby SDK is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with The Mercury Ruby SDK.  If not, see <http://www.gnu.org/licenses/>.
############################


# :title:System
# 
# The System class allows to manage system call object responses from the MoMS API (Mogreet Message System).
#
# The available modules are used to extend a Response object from the Mercury.
# 
# === Example of request to the Mercury::System:
# Using a Mercury object m.
# ==== Ping :
# ===== How to perform the 'ping' request:
#  p = m.system.ping
# ===== How to print its result:
#  puts p.get_response_message
# 
# == Contact
# 
# Author:: Thomas SCHWEBEL for Mogreet
# Website:: http://www.mogreet.com/
# Date:: Monday December 06, 2010
#


class Mercury::System
  
  def initialize client_id, token
    @argHash = { 'client_id' => client_id, 'token' => token }
  end
  
  ## SYSTEM OBJECT REQUESTS
  # The system object tests connectivity to MoMS API servers.	 
  
  # The ping method tests connectivity to the MoMS API servers.
  def ping args = {}

    # Creates and returns a Response extending the Ping module.
    Mercury.make_response_object("ping", args.merge(@argHash)).extend(Ping)
  end

  # The status method => Not available yet.
  def status
     raise NotImplementedError, "Status method not available yet"
  end
  
  ## END OF SYSTEM OBJECT REQUESTS
  
  ## MODULES
  
  # The Ping module implements methods for getting data from a Ping Response Object.
  # This module is the same as a basic Response Object.
  module Ping
    # Represents a ping response.
    # Same as a Response object.
  end
  
  # The System module implements methods for getting data from a System Response Object.
  # This module is not implemented yet.
  module Status
    # Not implemented yet.
  end
  
end

File: python/tutorials/tutorial_Python_2/SMSCury/system.py
#!/usr/bin/python
'''
  Copyright 2011 Julien Salvi for Mogreet
 
  This file is part of Mercury.
 
  Mercury is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.
 
  Mercury is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
  GNU General Public License for more details.
 
  You should have received a copy of the GNU General Public License
  along with Mercury. If not, see <http://www.gnu.org/licenses/>.
'''

import urllib
from xml.etree import ElementTree

class Ping(object):
    '''
    The Ping object contains the response from a ping request to the Moms API.
    '''    
    
    def __init__(self, urlPing):
        '''
        Initializes a new Ping response object.
        @param urlPing URL for calling requests from Mogreet API.
        '''
        
        #Parsing the xml document from the url
        try:
            xmlDoc = ElementTree.parse(urllib.urlopen(urlPing))  
        except IOError:
            print "Error while parsing the XML document"  
        
        root = xmlDoc.getroot()
        
        self.message = root.find(".//message").text
        self.responseCode = root.attrib.get("code")
        self.responseStatus = root.attrib.get("status")
             
    def __getattr__(self, attribut):
        print("Attribute {0} undefined!".format(attribut))
    
    def getMessage(self):
        '''
        @return: The message
        ''' 
        return self.message
            
    def getResponseStatus(self):
        '''
        @return: The response Status
        '''
        return self.responseStatus
       
    def getResponseCode(self):
        '''
        @return: The response Code
        ''' 
        return self.responseCode
    
    def responseIsValid(self):
        '''
        Checks if the response code is 1.
        @return: True if the response code is 1, False if it is not
        '''
        if (self.responseCode == 1):
            return True
        else:
            return False